wasmtime-jni/src/utils.rs

```diff
@@ -14,6 +14,19 @@ pub fn get_string_field(env: &JNIEnv, obj: JObject, field: &str) -> Result<Strin
     get_string(env, s)
 }
 
+/// Shorthand of obtaining and turning `JObject` from a field into a Rust `Option<String>`.
+pub fn get_optional_string_field(env: &JNIEnv, obj: JObject, field: &str) -> Result<Option<String>> {
```

Comment:
For values intended to be exchanged with JNI program, using just a nullable `String` is obviously easier than using `Optional`. Can you change the `WasiConfig` to hold `String` fields and add a function that checks `o.is_null()` and returns `Option<String>` instead?

Reply:
That makes sense 👍  I changed to use just a nullable `String` for simplexity.